### PR TITLE
cmake into ortoa client

### PR DESCRIPTION
**Pull request by adriandavila**
@jasondu7297 I don't quite understand why there's 26 commits listed but anyways the only 2 relevant ones are the last two. `d1b67e2` just merges `origin/ortoa-tee` into `ortoa-client`. `71e81d3` fixes all the relevant cmake things.

### CODE REVIEW

**Comment by jasondu7297** [comment]
> @jasondu7297 I don't quite understand why there's 26 commits listed but anyways the only 2 relevant ones are the last two. `d1b67e2` just merges `origin/ortoa-tee` into `ortoa-client`. `71e81d3` fixes all the relevant cmake things.

Oh my goodness... The way my jaw dropped when I saw the number of commits before your comment

File: data/sample_generated_seed/sample_operations.csv [hunk 1]
```diff
@@ -0,0 +1,100 @@
+GET,11
+PUT,375,975026
```
Comment (jasondu7297) [inline data/sample_generated_seed/sample_operations.csv:2]: Lets decrease the value size to <= 100000 as previously discussed. Values that are too large can risk decryption throwing an error which is sometimes seen in enclave.

File: data/sample_generated_seed/seed_data.csv [hunk 2]
```diff
@@ -0,0 +1,1000 @@
+PUT,1,636519
```
Comment (jasondu7297) [inline data/sample_generated_seed/seed_data.csv:1]: Same comment as above, let's generate data <= 100000.

**Review by jasondu7297 — APPROVED** [review]
Looks good, finished testing and all works out well. Some minor comments on value generation and include paths.

File: host/host.cpp [hunk 3]
```diff
@@ -6,80 +6,89 @@
...
 #include <thrift/server/TSimpleServer.h>
-#include <thrift/transport/TServerSocket.h>
 #include <thrift/transport/TBufferTransports.h>
+#include <thrift/transport/TServerSocket.h>
 
-#include "../errors/errors.h"
-#include "ortoa_u.h"
-#include "redis.h"
 #include "../constants/constants.h"
 #include "../constants/shared.h"
```
Comment (jasondu7297) [inline host/host.cpp:13]: One thing I noticed is that our includes sometimes try to use relative path and sometimes not (ex. `#include "../constants/shared.h"` vs  `#include "constants/shared.h"`). This was remnants of our old file directory structure + Makefile usage. Would it be possible to switch to one method or the other with CMake?

File: client/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,32 @@
+
+# Create the executable target
+add_executable(client 
+    client.cpp
+    client_utils.cpp
+    client_utils.h
+    ../host/redis.cpp 
+    ../crypto/encryption_engine.cpp
+    ../gen-cpp/RPC.h
```
Comment (jasondu7297) [inline client/CMakeLists.txt:9]: In the Host CMakeLists.txt, we do `${CMAKE_BINARY_DIR}/gen-cpp/...`. Here we use `../`. I think this was left from when I was fiddling with somethings lol but we should stay consistent and use one or the other

File: client/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    ../gen-cpp/RPC.cpp
+    ../gen-cpp/Operation_types.h
+    ../gen-cpp/Operation_types.cpp
+)
+
+add_dependencies(client generate-thrift)
+
+include_directories(
+    ${CMAKE_SOURCE_DIR}/constants
+    ../gen-cpp
```
Comment (jasondu7297) [inline client/CMakeLists.txt:19]: Are both `../gen-cpp` and `${CMAKE_BINARY_DIR}/gen-cpp` required?

File: host/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    --search-path ${OE_INCLUDEDIR} 
+    --search-path ${OE_INCLUDEDIR}/openenclave/edl/sgx)
+
+
+add_executable(ortoa-host
+               host.cpp 
+               redis.h
+               redis.cpp
+               ../constants/constants.h
+               ${CMAKE_BINARY_DIR}/gen-cpp/RPC.h
```
Comment (jasondu7297) [inline host/CMakeLists.txt:16]: Reference to comment https://github.com/ySteinhart1/ORTOA/pull/17/files#r1367831540

File: data/sample_generated_seed/sample_operations.csv (hunk 1 above)
Comment (adriandavila) [reply data/sample_generated_seed/sample_operations.csv:2]: I'll make this a separate PR

**Pull request merged**